### ISSUE TITLE
[Bug] Oddly specific problem with C-Buttons Y- not working with Right Analog Y- in a remapping.

### ISSUE THREAD

**Issue by iMeteox** [libretro/RetroArch#9815]
## Description

When creating a game remapping file for a Nintendo 64 game using Parallel core and a Gamecube controller (with vJoy), RetroArch will not properly handle the "C Button Y -" mapping. Please note that I have been able to reproduce this exact same issue on an Xbox One controller, so this problem is most likely not controller-related.

I remapped my controller in the following way;
C Buttons X+ = A button (right) 2 (N/A)
C Buttons X- = X button (top) 3 (N/A)
C Buttons Y+ = L button (shoulder) 4 (N/A)
C Buttons Y - =Right Analog Y- (up) -4 (N/A)

### Expected behavior

All the new mappings should work and I should be able to use my items in my Zelda roms using the X, Y and Z buttons of my gamecube controllers, while still using C-Stick (Right analog) up to look.

### Actual behavior

The C-Buttons mappings for X, Y and Z work as intended, but the C-stick up mapping doesn't work. Please, do note that;
1. I can map literally anything else than "C-buttons Y-" to my "Right analog Y-" and it will work.
2. I can map "C-buttons Y-" to literally anything else than "Right analog Y-" and it will work. 
3. If I map "C-buttons Y+" to "Right analog Y+", the "C-buttons Y-" binding will start working.

I have done that exact same remapping on my Xbox One controller and have met the exact same problem.

### Steps to reproduce the bug

1. Download and set up RetroArch following the LoZ - OoT randomizer guide : https://wiki.ootrandomizer.com/index.php?title=Retroarch
2. Set up your controller to the mappings described above
3. Start a game and try to use your C-Button up. I personally use a vanilla ROM of The Legend of Zelda - Ocarina of Time. I have tested on other ROMs and have met the same problem.

### Bisect Results

This as always happened for as long as I've tried to use this specific mapping.

### Version/Commit
You can find this information under Information/System Information

- RetroArch: 1.8.1
- Core: ParaLLEI N64 (2.0-rc2 fbf39c8)

### Environment information

- OS: Windows 10 Enterprise edition x64
- Compiler: No local build.

Here are included my remapping file and my RetroArch config file.

[LoZ - OoT (GZ).txt](https://github.com/libretro/RetroArch/files/3918498/LoZ.-.OoT.GZ.txt)
[RetroArch-1203-094217.txt](https://github.com/libretro/RetroArch/files/3918499/RetroArch-1203-094217.txt)

**Comment by bslenul** [comment]
Yup it's a bit weird, it's like "C-Buttons X/Y" controls are bound to analog stick, if you bind it to a normal button it's moving the right analog (you can see it with an overlay), but it's moving so slightly that it doesn't "activate" the input.

But anyway, here's a workaround (just tested myself, it work fine):

* `Quick Menu > Options > Independant C-buttons Controls` to "ON".
* `Quick Menu > Close Content` so the option saves properly and restart the game.
* `Quick Menu > Controls`:
    * bind A button to C-Right
    * X -> C-Down
    * L -> C-Left
    * and leave Right Analog Y+ to C Button Y -
* `Quick Menu > Resume` and check if it work.
* If everything is fine: `Quick Menu > Controls > Save Game Remap File` so you don't have to do this every time you boot the game.

**Comment by dobyrch** [comment]
I'm experiencing a similar bug: any buttons mapped to Right Analog Y (+ or -) do not work.  I first noticed it wasn't working while trying to play N64, but it seems to be broken on all cores.

A git bisect narrowed it down to 863901897638.

**Comment by rsn8887** [comment]
Can someone who knows this bug please confirm that PR #9840 fixes this? Thanks!

**Comment by bslenul** [comment]
Doesn't seem to change anything for me, however I noticed that it works sometimes (I mean even with latest commit retroarch.c).

In my case at least I noticed that it's because my controller has a dead zone issue, I made a video showing the issue: https://youtu.be/1sNQJ8rxGFY

I boot Ape Escape and I bound cross to right analog Y+ and square to right analog Y-, nothing seems to happen when I press cross/square, but with an overlay you can see the right stick barely moving (if you look close enough you can see it moving at ~2 seconds in the video), I set Analog Deadzone to 0.1 instead of 0.0 and \*tada\* it works just fine.

So yeah, idk about the other people having this issue but for me at least it seems to be a hardware issue, not RetroArch or cores' fault.

**Comment by dobyrch** [comment]
Thanks @rsn8887, your PR fixes the issue I was having.

However, now I've realize that the original bug reported by @iMeteox is *not* the same as my issue. That bug still remains (I can reproduce it with both Parallel and Mupen-Next) :

- "Right Analog X+" will not trigger  "C Buttons X+" unless "Right Analog X-" is also mapped to "C Buttons X-" (and vice versa)
- "Right Analog Y-" will not trigger  "C Buttons Y-" unless "Right Analog Y+" is also mapped to "C Buttons Y+"

Mapping any direction of the right analog stick to "C-{Up/Down/Left/Right}" Instead of "C Buttons {X/Y}{+/-}" works as expected.

Sorry for the mixup!

**Comment by rsn8887** [comment]
Hello @dobyrch , thanks for the testing. Is the original issue by @iMeteox that you reproduced in any way deadzone related?

**Comment by dobyrch** [comment]
No, it doesn't seem to be related to deadzones.  After applying your patch, I can still reproduce the original issue regardless of the value of `input_analog_deadzone`. 

It seems like the easiest solution would be to simply remove "C Buttons {X/Y} {+/-}" from the list of mappable inputs in the N64 cores—now that RetroArch supports mapping buttons to axes, it seems redundant to list C-axes *and* C-buttons, especially considering (1) the N64 controller doesn't even have "C-axes" and (2) mapping the C-buttons works just fine.

In any case, I'm satisfied with #9840, so I think I'll duck out of this issue for the time being.  If anything else comes up I'd be happy to test.

**Comment by rsn8887** [comment]
@dobyrch : I made a better fix without sacrificing radial analog stick deadzone, can you maybe test? https://github.com/libretro/RetroArch/pull/9842
